nbs_probabilistic/create_dropout_ensemble.py

```diff
@@ -0,0 +1,145 @@
+    
+    #ToDo: make loss function general.
+    #Break the code with error msg if not resnet or unet_google
+    #Important Ques: Since we dont build again, are we passing kernel, filters, activation, dropout and other details to the network?
+    if 'resnet' in exp_id:
+        model=tf.keras.models.load_model(saved_model_path
+                ,custom_objects={'PeriodicConv2D':PeriodicConv2D,'lat_mse': tf.keras.losses.mse})
+    elif 'unet_google' in exp_id:
+        model=tf.keras.models.load_model(saved_model_path, 
+                                         custom_objects={'PeriodicConv2D':PeriodicConv2D})#loss fn.?
+    else:
+        print('Only resnet or unet_google allowed')
+        #ToDO: Raise Exception
+        #Ques: Does Unet (not unet_google) work?
+    #model.summary()
+    return model
+
+def predict(dg_test,model,number_of_forecasts):
+    X,y=dg_test[0] #currently limiting output due to RAM issues.
+#     for i in range(len(dg_test)-1):
+#     X_batch,y_batch=dg_test[i+1]
+#     X=np.append(X,X_batch,axis=0)
+#     y=np.append(y,y_batch,axis=0)
+    
+    #test-time dropout
+    func = K.function(model.inputs + [K.learning_phase()], model.outputs)
+    pred_ensemble = np.array([np.asarray(func([X] + [1.]), dtype=np.float32).squeeze() for _ in
+                              range(number_of_forecasts)])
+    
+    #unnormalize
+    observation=y
+    pred_ensemble=(pred_ensemble * dg_test.std.isel(level=dg_test.output_idxs).values +
+         dg_test.mean.isel(level=dg_test.output_idxs).values)
+    observation=(observation * dg_test.std.isel(level=dg_test.output_idxs).values +
+         dg_test.mean.isel(level=dg_test.output_idxs).values)
+    
+    #numpy --> xarray. ToDo: change if output_vars is changed.
+    #convert from numpy to xarray
+    preds = xr.Dataset({
+    'z500': xr.DataArray(pred_ensemble[...,0],
+        dims=['forecast_number', 'time','lat', 'lon'],
+        coords={'forecast_number': np.arange(number_of_forecasts),'time': dg_test.data.time.isel(time=slice(None,X.shape[0])), 'lat': dg_test.data.lat, 'lon': dg_test.data.lon,},)
+    ,
+    't850': xr.DataArray(pred_ensemble[...,1],
+        dims=['forecast_number', 'time','lat', 'lon'],
+        coords={'forecast_number': np.arange(number_of_forecasts),'time': dg_test.data.time.isel(time=slice(None,X.shape[0])), 'lat': dg_test.data.lat, 'lon': dg_test.data.lon,},)
+})
+
+    observation= xr.Dataset({
+    'z500': xr.DataArray(observation[...,0],
+                         dims=['time','lat','lon'],
+                         coords={'time':dg_test.data.time.isel(time=slice(None,X.shape[0])),'lat':dg_test.data.lat,'lon':dg_test.data.lon},)
+    ,
+    't850': xr.DataArray(observation[...,1],dims=['time','lat','lon'],coords={'time':dg_test.data.time.isel(time=slice(None,X.shape[0])),'lat':dg_test.data.lat,'lon':dg_test.data.lon},)          
+})
+
+    return preds
+
+def main(exp_id_path,number_of_forecasts):
```

Comment:
As mentioned above, I think it would make sense to add `datadir` and `savedir` (where to save the predictions) as arguments here.